### DISCUSSION TITLE
Are PDFs edited on client side only ?

### DISCUSSION THREAD

**Discussion by guigui42** [Stirling-Tools/Stirling-PDF#174] [Q&A] [answered]
The broader question would be : when using stirling Pdf, are the PDFs file uploaded to the server side ? or is everything happening on the client side without data privacy issues ?

**Comment by Frooodle** [comment, marked as answer]
So it depends a lot of files are handled server side most of which the files are stored in memory meaning they are not even written to disk and go after request finished however some make temporary files instead during to process such as CLI ones.

Some features like the multi tool, rotate and sign are all clientside only

All files on server side are deleted before the request even finishes. No files are retained at all even for a single second after the request itself is finished
